projects/cds/src/profiletag/services/profile-tag-push-events.service.ts

```diff
@@ -0,0 +1,150 @@
+    this.searchResultsChanged(),
+    this.homePageVisitedEvent(),
+    this.cartPageVisitedEvent(),
+    this.pageVisitedEvent(),
+    this.orderConfirmationVisited(),
+    this.cartChanged()
+  );
+  constructor(
+    protected activeCartService: ActiveCartService,
+    protected eventService: EventService,
+    protected personalizationContextService: PersonalizationContextService
+  ) {}
+
+  getPushEvents(): Observable<ProfileTagPushEvent> {
+    return this.pushEvents$.pipe(
+      withLatestFrom(
+        merge(
+          of({ segments: undefined, actions: undefined }),
+          this.personalizationContextService.getPersonalizationContext()
+        )
+      ),
+      map(([item, personalizationContext]) => {
+        item.data = item.data ? item.data : {};
+        item.data.segments = personalizationContext.segments;
+        item.data.actions = personalizationContext.actions;
+        return item;
+      })
+    );
+  }
+
+  addPushEvent(event: Observable<ProfileTagPushEvent>) {
+    this.pushEvents$ = merge(this.pushEvents$, event);
+  }
+
+  /**
+   * Listens to the changes to the cart and pushes the event for profiletag to pick it up further.
+   */
+  cartChanged(): Observable<CartChangedPushEvent> {
+    return this.activeCartService.getActive().pipe(
+      skipWhile((cart) => !Boolean(cart.entries) || cart.entries.length === 0),
+      map(
+        (cart) =>
+          new CartChangedPushEvent({
+            cart,
+          })
+      )
+    );
+  }
+
+  protected categoryPageVisited(): Observable<ProfileTagPushEvent> {
+    return this.eventService.get(CategoryPageVisited).pipe(
+      map(
+        (categoryPageVisited) =>
+          new CategoryViewPushEvent({
+            productCategory: categoryPageVisited.categoryCode,
+            productCategoryName: categoryPageVisited.categoryName,
+          })
+      )
+    );
+  }
+
+  protected searchResultsChanged(): Observable<ProfileTagPushEvent> {
+    return this.eventService.get(KeywordSearchPageVisited).pipe(
+      map((searchEvent) => {
+        return new KeywordSearchPushEvent({
+          searchTerm: searchEvent.searchTerm,
+          numResults: searchEvent.numberOfResults,
+        });
+      })
+    );
+  }
+
+  protected productDetailsPageView(): Observable<ProfileTagPushEvent> {
+    return this.eventService.get(ProductDetailsPageVisited).pipe(
+      map(
+        (item) =>
+          new ProductViewPushEvent({
+            productSku: item.code,
+            productName: item.name,
+            productPrice: item.price ? item.price.value : undefined,
+            productCategoryName: item.categories
+              ? item.categories[item.categories.length - 1].name
+              : undefined,
+            productCategory: item.categories
+              ? item.categories[item.categories.length - 1].code
+              : undefined,
+          })
+      )
+    );
+  }
+
+  protected pageVisitedEvent(): Observable<ProfileTagPushEvent> {
```

Comment:
Are you sure you want to have this generic event here?
If we add other page visited events in the future, they might extend this one (or maybe not, see the comment in `projects/core/src/routing/event/routing.events.ts` about it). In that case you would get _FAQ_ page visited event, for example, and I'm not sure if it's useful to you?

Reply:
At the moment this is not useful, but I can imagine that as a next step we would replace our 'Navigated' with the 'PageVisited'. But we can remove this for now until we need it